### PR TITLE
Changing license for application-mobility

### PR DESCRIPTION
**Pull request by abhi16394**
# Description
New license added for application-mobility and removed the variable  "APPLICATION_MOBILITY_LICENSE_NAME" from sample files and other testing files.
There is a new method to generate secret and wouldn't require the secret "license" to be created anymore. 

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [ ] Ran unit-test
- [ ] Ran backup + restore with deployed AM + driver CSM, see below for details
```
[root@master-1-anCjwbyux3TqW csm-operator]# kubectl create -f samples/application-mobility/csm_application_mobility_v030.yaml
```
`the pods come up:

```
[root@master-1-anCjwbyux3TqW ~]# kubectl get pods -n vxflexos
NAME                                                       READY   STATUS    RESTARTS   AGE
application-mobility-controller-manager-67c668477f-dpn82   2/2     Running   0          15m
application-mobility-node-agent-fzjfs                      1/1     Running   0          15m
application-mobility-node-agent-pf4cq                      1/1     Running   0          15m
application-mobility-velero-55cf6cc5c4-xnggn               1/1     Running   0          15m
cert-manager-74b9fc6d95-2xrss                              1/1     Running   0          16m
cert-manager-cainjector-7bc5c4fd45-9xfrr                   1/1     Running   0          16m
cert-manager-webhook-85b4fd46d9-gwf52                      1/1     Running   0          16m
vxflexos-app-mobility-controller-cc86ff647-gqtrm           5/5     Running   0          15m
vxflexos-app-mobility-node-97sgc                           2/2     Running   0          15m
vxflexos-app-mobility-node-vq9m5                           2/2     Running   0          15m


```

CSM status:

```
[root@master-1-anCjwbyux3TqW ~]# kubectl get csm -A
NAMESPACE   NAME                    CREATIONTIME   CSIDRIVERTYPE   CONFIGVERSION   STATE
vxflexos    vxflexos-app-mobility   17m            powerflex       v2.8.0          Succeeded

```
Deploy Wordpress App:
```
kubectl apply -n wordpress -k sample-application
```

wordpress pods :
```
[root@master-1-CMyNgvaXZTEw3 e2e]# kubectl get pods -n wordpress
NAME                               READY   STATUS    RESTARTS   AGE
wordpress-5584ff7f88-v6ws2         1/1     Running   0          103s
wordpress-mysql-664b79468d-skrz9   1/1     Running   0          103s
```
create backup :
```
[root@master-1-CMyNgvaXZTEw3 bin]# ./dellctl backup create backup1 --include-namespaces wordpress -n vxflexos
```
confirm backup is created:
```
[root@master-1-CMyNgvaXZTEw3 bin]# ./dellctl backup get -n vxflexos
NAME      STATUS      CREATED                         EXPIRES                         STORAGE LOCATION   DATA MOVER   CLONED   TARGET CLUSTERS
backup1   Completed   2023-09-21 18:03:28 -0400 EDT   2023-10-21 18:03:28 -0400 EDT   default            Restic       false

```
create restore:
```
./dellctl restore create restore1 --from-backup backup1 -n vxflexos --namespace-mappings wordpress:res-wordpress
```
confirm if restore is created:
```
[root@master-1-CMyNgvaXZTEw3 bin]# ./dellctl restore get -n vxflexos
NAME       BACKUP    STATUS      CREATED                         COMPLETED
restore1   backup1   Completed   2023-09-21 18:07:54 -0400 EDT   NA
```
Verify if pods have come up in restore namespace:
```
[root@master-1-CMyNgvaXZTEw3 bin]# kubectl get pods -n res-wordpress
NAME                               READY   STATUS    RESTARTS   AGE
wordpress-5584ff7f88-zdsct         1/1     Running   0          3m14s
wordpress-mysql-664b79468d-mrb44   1/1     Running   0          3m14s
```

### CODE REVIEW

File: deploy/operator.yaml [hunk 1]
```diff
@@ -1251,7 +1251,7 @@ spec:
         - --leader-elect
         command:
         - /manager
-        image: docker.io/dellemc/dell-csm-operator:v1.3.0
+        image: 10.247.98.98:5000/csm:lic01
```
Comment (JacobGros) [inline deploy/operator.yaml:1254]: we should take out internal IPs

File: operatorconfig/moduleconfig/application-mobility/v0.3.0/app-mobility-controller-manager.yaml [hunk 2]
```diff
@@ -36,7 +36,7 @@ spec:
         - --leader-elect
         - --app-mobility-namespace=<NAMESPACE>
         - --secret-name=<APPLICATION_MOBILITY_OBJECT_STORE_SECRET_NAME>
-        - --license-name=<APPLICATION_MOBILITY_LICENSE_NAME>
+        #- --license-name=<APPLICATION_MOBILITY_LICENSE_NAME>
```
Comment (JacobGros) [inline operatorconfig/moduleconfig/application-mobility/v0.3.0/app-mobility-controller-manager.yaml:39]: can you remove the commented out line?

File: pkg/modules/application_mobility.go [hunk 3]
```diff
@@ -399,10 +392,11 @@ func ApplicationMobilityPrecheck(ctx context.Context, op utils.OperatorConfig, a
 	}
 
 	// Check for secrets
-	appMobilitySecrets := []string{"license"}
+	ns := "default"
+	appMobilitySecrets := []string{"dls-license", "iv"}
 	for _, name := range appMobilitySecrets {
 		found := &corev1.Secret{}
-		err := r.GetClient().Get(ctx, types.NamespacedName{Name: name, Namespace: cr.GetNamespace()}, found)
+		err := r.GetClient().Get(ctx, types.NamespacedName{Name: name, Namespace: ns}, found)
```
Comment (JacobGros) [inline pkg/modules/application_mobility.go:399]: do we know DL license is always in default NS?
Could be a question for @harshitap26

Comment (JacobGros) [reply pkg/modules/application_mobility.go:399]: same question about license name

File: samples/application-mobility/csm_application_mobility_v030.yaml [hunk 4]
```diff
@@ -15,7 +15,7 @@ spec:
       - name: application-mobility-controller-manager
         # enable: Enable/Disable application mobility controller-manager
         enabled: true
-        image: dellemc/csm-application-mobility-controller:v0.3.0
+        image: amaas-eos-mw1.cec.lab.emc.com:5098/csm-application-mobility:latest
```
Comment (JacobGros) [inline samples/application-mobility/csm_application_mobility_v030.yaml:18]: Internal image name shouldn't be in repo

**Review by JacobGros — COMMENTED** [review]
Can you try running the full backup and restore test with dellctl?
Process was documented in "How Has This Been Tested?"
section here: https://github.com/dell/csm-operator/pull/358

File: pkg/modules/application_mobility.go (hunk 3 above)
Comment (harshitap26) [reply pkg/modules/application_mobility.go:399]: @JacobGros Yes, right now the licensing service is creating the dls-license and iv secret in the default namespace.

File: deploy/operator.yaml (hunk 1 above)
Comment (abhi16394) [reply deploy/operator.yaml:1254]: changed it

File: operatorconfig/moduleconfig/application-mobility/v0.3.0/app-mobility-controller-manager.yaml (hunk 2 above)
Comment (abhi16394) [reply operatorconfig/moduleconfig/application-mobility/v0.3.0/app-mobility-controller-manager.yaml]: removed it

File: samples/application-mobility/csm_application_mobility_v030.yaml (hunk 4 above)
Comment (abhi16394) [reply samples/application-mobility/csm_application_mobility_v030.yaml]: changed it

File: samples/application-mobility/csm_application_mobility_with_driver.yaml [hunk 5]
```diff
@@ -471,16 +466,16 @@ spec:
...
+              aws_access_key_id:  #Provide the access key id here
+              aws_secret_access_key:  #provide the access key here 
               
 
     # Init containers to be added to the Velero and Node-agent deployment's spec.
     - initContainer:
         #initContainer image for the dell velero plugin
       - name: dell-custom-velero-plugin
-        image:
+        image:
```
Comment (kumarkgosa) [inline samples/application-mobility/csm_application_mobility_with_driver.yaml:477]: Why did we remove plugin images here ?

**Review by kumarkgosa — APPROVED** [review]

File: samples/application-mobility/csm_application_mobility_with_driver.yaml (hunk 5 above)
Comment (abhi16394) [reply samples/application-mobility/csm_application_mobility_with_driver.yaml:477]: it was the same before

Comment (JacobGros) [reply samples/application-mobility/csm_application_mobility_with_driver.yaml:477]: yeah, there is no public plugin image to use
We will fill it in when we have it posted on DockerHub

**Pull request merged**